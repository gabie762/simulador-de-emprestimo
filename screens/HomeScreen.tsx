import React from 'react';
import { Text, TouchableOpacity, View } from 'react-native';
import { globalStyles } from '../styles';

interface HomeScreenProps {
  onNavigate: (screen: string) => void;
}

const HomeScreen: React.FC<HomeScreenProps & { testID?: string }> = ({ onNavigate, testID }) => {
  return (
    <View style={globalStyles.container} testID={testID}>
      <Text style={globalStyles.title}>Simulador de Empréstimos</Text>
      <Text style={globalStyles.subtitle}>O que você gostaria de fazer?</Text>
      <TouchableOpacity 
        style={globalStyles.button} 
        onPress={() => onNavigate('cadastro')}
        testID="navigate-cadastro"
        accessibilityLabel="Cadastrar Produto"
      >
        <Text style={globalStyles.buttonText}>Cadastrar Produto</Text>
      </TouchableOpacity>
      <TouchableOpacity 
        style={globalStyles.button} 
        onPress={() => onNavigate('listaProdutos')}
        testID="navigate-listaProdutos"
        accessibilityLabel="Listar Produtos"
      >
        <Text style={globalStyles.buttonText}>Listar Produtos</Text>
      </TouchableOpacity>
      <TouchableOpacity 
        style={globalStyles.button} 
        onPress={() => onNavigate('simulacao')}
        testID="navigate-simulacao"
        accessibilityLabel="Nova Simulação"
      >
        <Text style={globalStyles.buttonText}>Nova Simulação</Text>
      </TouchableOpacity>
    </View>
  );
};

export default HomeScreen;
