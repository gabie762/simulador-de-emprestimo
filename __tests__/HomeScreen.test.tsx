import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import HomeScreen from '../screens/HomeScreen';

describe('HomeScreen', () => {
  it('renders correctly', () => {
    const mockOnNavigate = jest.fn();
    const { toJSON } = render(<HomeScreen onNavigate={mockOnNavigate} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onNavigate with "cadastro" when the Cadastrar Produto button is pressed', () => {
    const mockOnNavigate = jest.fn();
    const { getByText } = render(<HomeScreen onNavigate={mockOnNavigate} />);

    fireEvent.press(getByText('Cadastrar Produto'));

    expect(mockOnNavigate).toHaveBeenCalledWith('cadastro');
  });

  it('calls onNavigate with "listaProdutos" when the Listar Produtos button is pressed', () => {
    const mockOnNavigate = jest.fn();
    const { getByText } = render(<HomeScreen onNavigate={mockOnNavigate} />);

    fireEvent.press(getByText('Listar Produtos'));

    expect(mockOnNavigate).toHaveBeenCalledWith('listaProdutos');
  });

  it('calls onNavigate with "simulacao" when the Nova Simulação button is pressed', () => {
    const mockOnNavigate = jest.fn();
    const { getByText } = render(<HomeScreen onNavigate={mockOnNavigate} />);

    fireEvent.press(getByText('Nova Simulação'));

    expect(mockOnNavigate).toHaveBeenCalledWith('simulacao');
  });

  it('has accessible buttons', () => {
    const mockOnNavigate = jest.fn();
    const { getByLabelText } = render(<HomeScreen onNavigate={mockOnNavigate} />);

    expect(getByLabelText('Cadastrar Produto')).toBeTruthy();
    expect(getByLabelText('Listar Produtos')).toBeTruthy();
    expect(getByLabelText('Nova Simulação')).toBeTruthy();
  });
});
