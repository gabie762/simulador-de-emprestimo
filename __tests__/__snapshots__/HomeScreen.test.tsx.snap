// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "flex-start",
      "maxWidth": 600,
      "padding": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#0066B3",
        "fontFamily": "Caixa",
        "fontSize": 28,
        "fontWeight": "bold",
        "marginBottom": 20,
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    Simulador de Empréstimos
  </Text>
  <Text
    style={
      {
        "color": "#F7941E",
        "fontFamily": "Caixa",
        "fontSize": 18,
        "marginBottom": 40,
        "textAlign": "center",
      }
    }
  >
    O que você gostaria de fazer?
  </Text>
  <View
    accessibilityLabel="Cadastrar Produto"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#0066B3",
        "borderRadius": 8,
        "marginBottom": 10,
        "opacity": 1,
        "padding": 15,
        "width": "100%",
      }
    }
    testID="navigate-cadastro"
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "Caixa",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Cadastrar Produto
    </Text>
  </View>
  <View
    accessibilityLabel="Listar Produtos"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#0066B3",
        "borderRadius": 8,
        "marginBottom": 10,
        "opacity": 1,
        "padding": 15,
        "width": "100%",
      }
    }
    testID="navigate-listaProdutos"
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "Caixa",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Listar Produtos
    </Text>
  </View>
  <View
    accessibilityLabel="Nova Simulação"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#0066B3",
        "borderRadius": 8,
        "marginBottom": 10,
        "opacity": 1,
        "padding": 15,
        "width": "100%",
      }
    }
    testID="navigate-simulacao"
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "Caixa",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Nova Simulação
    </Text>
  </View>
</View>
`;
